lib/iris/analysis/maths.py

```diff
@@ -27,9 +28,7 @@
 from dask.array.core import broadcast_shapes
 
 
-_output_dtype_cache = {}
-
-
+@lru_cache(maxsize=128, typed=True)
```

Comment:
Since 128 is the default value ([docs](https://docs.python.org/3/library/functools.html)), can this be:
```python
@lru_cache(typed=True)
```

Reply:
@trexfeathers Reasonable request, but I'd rather be explicit here, thanks.